src/component/BpmnVisualization.ts

```diff
@@ -92,4 +317,171 @@ export class BpmnVisualization {
+      console.log(filter);
+
+      if (filter.length >= 1) {
+        // See https://github.com/jgraph/mxgraph/blob/ff141aab158417bd866e2dfebd06c61d40773cd2/javascript/src/js/view/mxGraph.js#L2114
+        this.undoManager.registerUndoable((<mxStyleChange>filter[0]).cell, edit);
+      }
+    });
+
+    // Mandatory
+    this.graph.getModel().addListener(mxgraph.mxEvent.UNDO, listener);
+    this.graph.getView().addListener(mxgraph.mxEvent.UNDO, listener);
+  }
+
+  updateStyle<C extends string>(bpmnElementIds: string | string[], style: ShapeStyleUpdate<C> | EdgeStyleUpdate<C>): void {
+    this.graph.getModel().beginUpdate();
+    try {
+      ensureIsArray<string>(bpmnElementIds).forEach(bpmnElementId => {
+        const cell = this.graph.getModel().getCell(bpmnElementId);
+        let cellStyle = cell.getStyle();
+
+        console.log('old style = %s', cellStyle);
+        console.log(`---- new style -----`);
+        console.log(style);
+
+        cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_OPACITY, style.opacity);
+        cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_SHADOW, Number(style.withShadow));
+
+        const font = style.font;
+        if (font) {
+          cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_FONTCOLOR, font.color);
+          cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_FONTSIZE, font.size);
+          cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_FONTFAMILY, font.family);
+          cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_FONTSTYLE, BpmnVisualization.getFontStyleValue(font));
+          cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_TEXT_OPACITY, font.opacity);
+        }
+
+        const label = style.label;
+        if (label) {
+          cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_LABEL_WIDTH, label.width);
+          cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_LABEL_POSITION, label.horizontalPosition);
+          cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_VERTICAL_LABEL_POSITION, label.verticalPosition);
+          cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_LABEL_BACKGROUNDCOLOR, label.backgroundColor);
+          cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_LABEL_BORDERCOLOR, label.borderColor);
+          cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_LABEL_PADDING, label.padding);
+        }
+
+        const stroke = style.stroke;
+        if (stroke) {
+          cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_STROKECOLOR, stroke.color);
+          cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_STROKEWIDTH, stroke.width);
```

Comment:
**chore:** 
We must not override this property, because we can have a conflict with the BPMN spec.